Bluetooth: Define connection event nofitication callbacks (#16099)
nrfconnect/sdk-nrf: samples/bluetooth/conn_evt_cb/README.rst

.. _ble_conn_evt_cb:

Bluetooth: Connection event callback
#####################################

.. contents::
   :local:
   :depth: 2

> The Connection event callback sample demonstrates how to use the connection event prepare callback feature described in :ref:`ug_bt_conn_evt_cb`.

Reviewer: ```suggestion
This sample demonstrates how to use the :ref:`ug_bt_conn_evt_cb` feature.
```

> It uses the :ref:`latency_readme` and the :ref:`latency_client_readme` to showcase how this feature can be used to minimize the time between data sampling and data transmission.

Reviewer: ```suggestion
It uses the :ref:`latency_readme` and the :ref:`latency_client_readme` to showcase how you can use this feature to minimize the time between data sampling and data transmission.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

You can use any two of the development kits mentioned above and mix different development kits.

Additionally, the sample requires a connection to a computer with a serial terminal for each of the development kits.

> Note that the feature involves triggering a (D)PPI task directly from the SoftDevice Controller link layer, and therefore it is expected that the application and SoftDevice Controller are running on the same core.

Reviewer: ```suggestion
.. note::
   The feature involves triggering a (D)PPI task directly from the SoftDevice Controller link layer.
   The application and SoftDevice Controller are supposed to be running on the same core.
```

Building and running
********************
.. |sample path| replace:: :file:`samples/bluetooth/llpm`

.. include:: /includes/build_and_run.txt

Testing
=======

> After programming the sample to both development kits, test it by performing the following steps:

Reviewer: ```suggestion
After programming the sample to both development kits, perform the following steps to test it:
```

1. Connect to both kits with a terminal emulator (for example, `nRF Connect Serial Terminal`_).
   See :ref:`test_and_optimize` for the required settings and steps.
#. Reset both kits.

> #. In one of the terminal emulators, type "c" to start the application on the connected board in the central role.
> #. In the other terminal emulator, type "p" to start the application in the peripheral role.

Reviewer: ```suggestion
#. In one of the terminal emulators, type ``c`` to start the application on the connected board in the central role.
#. In the other terminal emulator, type ``p`` to start the application in the peripheral role.
```

#. Observe that latency measurements are printed in the terminals.

Sample output
=============

The result should look similar to the following output.

> - For the central::

Reviewer: ```suggestion
For the central::
```

   I: SoftDevice Controller build revision:
   I: d6 da c7 ae 08 db 72 6f |......ro
   I: 2a a3 26 49 2a 4d a8 b3 |*.&I*M..
   I: 98 0e 07 7f             |....
   I: HW Platform: Nordic Semiconductor (0x0002)
   I: HW Variant: nRF52x (0x0002)
   I: Firmware: Standard Bluetooth controller (0x00) Version 214.51162 Build 1926957230
   I: Identity: FA:BB:79:57:D6:45 (random)
   I: HCI: version 5.4 (0x0d) revision 0x11fb, manufacturer 0x0059
   I: LMP: version 5.4 (0x0d) subver 0x11fb
   Choose device role - type c (central) or p (peripheral):
   Central. Starting scanning
   Scanning started
   Device found: CF:99:32:A5:4B:11 (random) (RSSI -43)
   Connected: CF:99:32:A5:4B:11 (random)
   Service discovery completed
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us
   Latency: 1116 us, round trip latency: 51116 us

> - For the peripheral::

Reviewer: ```suggestion
For the peripheral::
```
